app/src/components/screens/Notifications/Notification.js

```diff
@@ -8,23 +8,34 @@ import store from "../../../store/store";
 import { redirectUserToContext } from "../../../services/notifications/pushNotifications";
 
 const NotificationDetail = ({ notification }) => {
+  const [isRedirecting, setIsRedirecting] = useState(false);
+
   const { code, body, date, viewedAt, action } = notification;
 
   const ago = moment(date).fromNow(true);
 
-  const onPressNotification = () => {
-    if (action) {
-      redirectUserToContext(action.routeName, action.params, null, store);
-    }
+  const onPressNotification = async () => {
+    setIsRedirecting(true);
+    await redirectUserToContext(action.routeName, action.params, null, store);
+    setIsRedirecting(false);
   };
 
   return (
     <TouchableOpacity
       onPress={onPressNotification}
+      disabled={!action}
       style={{
         ...styles.container,
         backgroundColor: viewedAt ? colors.lightGray : colors.white,
       }}>
+      {isRedirecting ? (
```

Comment:
Si es un boolean, podes hacer lo siguiente:
```suggestion
      {isRedirecting && (
          <ActivityIndicator
            size="large"
            color={colors.mainBlue}
            style={styles.loading}
          />
       )
```
Y te evitas tener que decirle que hacer en caso de false